code/parachain/runtime/common/src/xcmp/mod.rs

```diff
@@ -94,6 +95,26 @@ pub struct TransactionFeePoolTrader<
 	asset_location: Option<MultiLocation>,
 }
 
+impl<
+		AssetConverter: Convert<MultiLocation, Option<CurrencyId>>,
+		PriceConverter: MinimalOracle<AssetId = CurrencyId, Balance = Balance>,
+		Treasury: TakeRevenue,
+		WeightToFeeConverter: WeightToFeePolynomial<Balance = Balance> + WeightToFee<Balance = Balance>,
+	> TransactionFeePoolTrader<AssetConverter, PriceConverter, Treasury, WeightToFeeConverter>
+{
+	pub fn weight_to_asset(
+		weight: Weight,
+		asset_id: CurrencyId,
+	) -> Result<(Balance, Balance), XcmError> {
+		let fee = WeightToFeeConverter::weight_to_fee(&weight);
+		log::trace!(target : "xcmp::weight_to_asset", "required payment in native token is: {:?}", fee );
+		let price =
+			PriceConverter::get_price_inverse(asset_id, fee).map_err(|_| XcmError::TooExpensive)?;
+		let price = price.max(Balance::one());
```

Comment:
NIT:
Why one as opposed to ED or another value? Should this be configurable?

Reply:
for high value low decimals currencies when our honest fee less than 1